### PR TITLE
Add body parser for HttpRequest

### PR DESCRIPTION
**Pull request by WitPxxl**
In case of an api, we need to get the body of request.
The body with php://input is readable only one time. So a parser to have this body multiple time can be useful.
Add an array to add multiple parsers if needed.

### CODE REVIEW

**Comment by gauthier** [comment]
Hi @witpxxl,

Thanks for your contribution on message. We’re currently evaluating whether we can merge your PR as is or if we should consider implement the feature you need differently. 

Either way, we’ll manage to handle this quickly. Thanks for being some more patient :)

**Comment by rflavien** [comment]
Hi, according to me it's `ObjectivePHP\Application\Operation\RequestWrapper` responsibility to hydrate the `parsedBody` property. Making the `getParsedBody` method a simple property accessor.

File: src/Request/HttpRequest.php [hunk 1]
```diff
@@ -1,193 +1,271 @@
...
-        /**
-         * @param HttpParameterContainer $parameters
-         */
-        public function setParameters(ParameterContainerInterface $parameters)
-        {
-            $this->parameters = $parameters;
+    /**
+     * @var bool|array
+     */
+    protected $bodyParsed = false;
```
Comment (rflavien) [inline src/Request/HttpRequest.php]: For consistency it should be `parsedBody`

**Comment by WitPxxl** [comment]
I put this method like that in the HttpRequest according to the PSR-7 ServerRequestInterface. It's inspired by SlimFramework. But i see also what you mean.

**Comment by rflavien** [comment]
@WitPxxl changing the extends `Psr\Http\Message\Request` by `Psr\Http\Message\ServerRequest` and using the `Zend\Diactoros\ServerRequestFactory` in `ObjectivePHP\Application\Operation\RequestWrapper` will reach the same goal with less code to maintain.
I think it will not break things as `ServerRequest` extends `Request`.

**Comment by WitPxxl** [comment]
@rflavien So I update `RequestWrapper` in application and adapt `HttpRequest` as your suggest or you will do this job start from my PR ?
I think it will not have less code to maintain. `Zend\Diactoros\ServerRequestFactory` doesn't parse the body in the parsedBody if i'm not wrong. We need to move the parser from Request to the Wrapper ?
Thanks

**Comment by rflavien** [comment]
Well, as request body can be anything, I think parsing is user land responsibility and should not be done by the framework.
You can easily create a middleware to make your needed parsing (or extend the `RequestWrapper`).

**Pull request closed without merging**